### PR TITLE
add user per period query scripts

### PR DESCRIPTION
**Pull request by szd55gds**
What
These scripts query for a set of matching user IDs per period and pull the IDs to local files.

Why
We will use the IDs to query tickets and last login data to remove IDs active > 1 year ago for GDPR.

### CODE REVIEW

File: lib/count-users-by-year-improved.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+today = Date.today.prev_day
+lastyear = today - 365
+
+#  puts "Total Tickets: #{@client.users.count}"
+
+puts "Total Users 2012"
+puts @client.search(:query => "type:user created_at>2012-01-01 created_at<2013-01-01 role:end-user -name:Zendesk organization_id:none").count
+puts "Total Users 2013"
+puts @client.search(:query => "type:user created_at>2013-01-01 created_at<2014-01-01 role:end-user -name:Zendesk organization_id:none").count
```
Comment (Krenair) [inline lib/count-users-by-year-improved.rb]: It looks like we're going to miss all users created on new year's day. And 2017-04-01 and 2017-07-01.

Comment (Krenair) [reply lib/count-users-by-year-improved.rb]: Turns out this is a datetime field so we'd probably only be missing stuff created at exactly 00:00:00

**Comment by szd55gds** [comment]
> Let's not introduce any more code duplicated by year. See #5

That would be nice but no time left to make pretty scripts and skill gap with ruby, so this is to get the job done in limited hours remaining. I agree in an ideal world.

**Comment by szd55gds** [comment]
Closed in error.

**Pull request merged**